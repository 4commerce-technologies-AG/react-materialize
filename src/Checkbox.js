import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import idgen from './idgen';

class Checkbox extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const { indeterminate } = this.props;

    if (this._input) {
      this._input.indeterminate = indeterminate;
      this._input.checked = false;
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.indeterminate !== this.props.indeterminate) {
      this._input.indeterminate = this.props.indeterminate;
    }
  }

  render() {
    const {
      id,
      indeterminate,
      filledIn,
      disabled,
      onChange,
      checked,
      value,
      label
    } = this.props;

    let computedId = id || idgen();

    if (indeterminate) {
      computedId = `indeterminate-checkbox-${idgen()}`;
    }

    return (
      <label htmlFor={computedId}>
        <input
          id={computedId}
          className={cx({
            'filled-in': filledIn
          })}
          disabled={disabled}
          onChange={onChange}
          type="checkbox"
          defaultChecked={checked}
          value={value}
          ref={input => (this._input = input)}
        />
        <span>{label}</span>
      </label>
    );
  }
}

Checkbox.propTypes = {
  /*
   * checkbox value
   */
  value: PropTypes.string.isRequired,
  /*
   * filled-in styled checkbox
   */
  filledIn: PropTypes.bool,
  /*
   * label next to checkbox
   */
  label: PropTypes.string.isRequired,
  /*
   * Indeterminate Style
   */
  indeterminate: PropTypes.bool,
  /*
   * onChange callback
   */
  onChange: PropTypes.func,
  /*
   * override id
   * @default idgen()
   */
  id: PropTypes.string,
  /*
   * disabled input
   */
  disabled: PropTypes.bool,
  /*
   * initialise checkbox checked
   */
  checked: PropTypes.bool
};

export default Checkbox;
