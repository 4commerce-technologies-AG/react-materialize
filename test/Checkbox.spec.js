import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../src/Checkbox';

describe('<Checkbox />', () => {
  let wrapper;

  test('renders', () => {
    wrapper = shallow(<Checkbox value="red" label="red" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders disabled', () => {
    wrapper = shallow(<Checkbox value="red" label="red" disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  test('filled-in', () => {
    wrapper = shallow(<Checkbox value="red" label="red" filledIn />);
    expect(wrapper).toMatchSnapshot();
  });
});
