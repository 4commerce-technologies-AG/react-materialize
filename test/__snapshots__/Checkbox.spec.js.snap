// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> does not overwrite materialize-css classnames 1`] = `
<Checkbox
  className="my-class"
  filledIn={true}
  id="Checkbox_mockId"
  label="red"
  value="red"
>
  <label
    htmlFor="Checkbox_mockId"
  >
    <input
      className="filled-in my-class"
      id="Checkbox_mockId"
      onChange={[Function]}
      type="checkbox"
      value="red"
    />
    <span>
      red
    </span>
  </label>
</Checkbox>
`;

exports[`<Checkbox /> filled-in 1`] = `
<label
  htmlFor="Checkbox_mockId"
>
  <input
    className="filled-in"
    id="Checkbox_mockId"
    onChange={[Function]}
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;

exports[`<Checkbox /> renders 1`] = `
<label
  htmlFor="Checkbox_mockId"
>
  <input
    className=""
    id="Checkbox_mockId"
    onChange={[Function]}
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;

exports[`<Checkbox /> renders disabled 1`] = `
<label
  htmlFor="Checkbox_mockId"
>
  <input
    className=""
    disabled={true}
    id="Checkbox_mockId"
    onChange={[Function]}
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;

exports[`<Checkbox /> with id 1`] = `
<label
  htmlFor="test"
>
  <input
    className=""
    id="test"
    onChange={[Function]}
    type="checkbox"
    value="red"
  />
  <span>
    red
  </span>
</label>
`;
